### ISSUE TITLE
Add Argparser type casting

### ISSUE THREAD

**Issue by Sygmei** [Sygmei/ObEngine#449]
The argparser is located here : https://github.com/ObEngine/ObEngine/blob/v0.5/src/Core/Utils/ArgParser.cpp#L7

The improvements would be the following one :
- Detect numbers (int and float) and cast them to appropriate vili type
- Detect `true` and `false` and cast them to booleans
- Detect flags without values (`--enable-this --enable-that`) and cast them to `true` (boolean)
- Quoting a value will force it to be a string but non-quoted values are also fine (except of course if you want a string equal to "true" for example)

**Comment by Sygmei** [comment]
@DarthData410 new issue created :)

**Comment by DarthData410** [comment]
Great @Sygmei assign me this issue, as I will take care of this.

**Comment by DarthData410** [comment]
Alright so I have the following Lua script in the Hello World project.:
```lua
function Event.Game.Render()
    SampleObject.components.Sprite:set_texture(_canvas:render());
    print(Engine.Args.arg);
    print(type(Engine.Args.arg));
end
```
With that if I run:
> ./ObEnginePlayer --arg TrUe

I get the following output:
> true
> boolean

if I run:
> ./ObEnginePlayer --arg 123

I get the following output:
> 123
> number

if I run:
> ./ObEnginePlayer --arg 3.14

I get the following output:
> 3.14
> number

if I run:
> ./ObEnginePlayer --arg 3.14f

I get the following output:
> 3.14f
> string

If I run:
 > ./ObEnginePlayer --arg "a string value"

I get the following output:
> a string value
> string

I am about to commit the changes and create a PR. I wanted to get this level of args parser in place, and then we can look at the --enable-somearg.

**Comment by DarthData410** [comment]
I wanted to complete a final test for this. I modified the sample project, to the following, and when a Engine.Args.arg is used, that is rendered on the screen correctly.
```lua
---@class SampleObject : GameObjectCls
local SampleObject = GameObject();

local _canvas;

function SampleObject:init()
    local screen_size = Engine.Window:get_size();
    _canvas = obe.canvas.Canvas(screen_size.x, screen_size.y);

    _canvas:Text "argrenderer" {
        text = tostring(Engine.Args.arg),
        size = 128,
        color = obe.graphics.Color.White,
        x = 0.5,
        y = 0.5,
        unit = obe.transform.Units.ViewPercentage,
        align = {
            horizontal = "Center",
            vertical = "Center"
        }
    };
end

function Event.Game.Render()
    SampleObject.components.Sprite:set_texture(_canvas:render());
    print(Engine.Args.arg);
    print(type(Engine.Args.arg));
end
```
> Note the **text = tostring(Engine.Args.arg)**

**Comment by DarthData410** [comment]
@Sygmei this issue can be closed based on pr #450
